src/transformers/models/superpoint/modeling_superpoint.py

```diff
@@ -0,0 +1,583 @@
+            batched_last_hidden_state = batched_encoder_outputs[0]
+            if output_hidden_states:
+                batched_hidden_states = batched_encoder_outputs[1]
+        else:
+            batched_last_hidden_state = batched_encoder_outputs.last_hidden_state
+            if output_hidden_states:
+                batched_hidden_states = batched_encoder_outputs.hidden_states
+
+        listed_keypoints_scores = [
+            self.keypoint_decoder(last_hidden_state[None, ...]) for last_hidden_state in batched_last_hidden_state
+        ]
+
+        listed_keypoints, listed_scores = [keypoints_scores[0] for keypoints_scores in listed_keypoints_scores], [
+            keypoints_scores[1] for keypoints_scores in listed_keypoints_scores
+        ]
+
+        listed_descriptors = [
+            self.descriptor_decoder(last_hidden_state[None, ...], keypoints[None, ...])
+            for last_hidden_state, keypoints in zip(batched_last_hidden_state, listed_keypoints)
+        ]
+
+        maximum_num_keypoints = max([keypoints.shape[0] for keypoints in listed_keypoints])
+
+        batched_keypoints = torch.zeros((batch_size, maximum_num_keypoints, 2), device=pixel_values.device)
+        batched_scores = torch.zeros((batch_size, maximum_num_keypoints), device=pixel_values.device)
+        batched_descriptors = torch.zeros(
+            (batch_size, maximum_num_keypoints, self.config.descriptor_dim),
+            device=pixel_values.device,
+        )
+        batched_mask = torch.zeros((batch_size, maximum_num_keypoints), device=pixel_values.device, dtype=torch.int)
+
+        for i, (keypoints, scores, descriptors) in enumerate(zip(listed_keypoints, listed_scores, listed_descriptors)):
+            batched_keypoints[i, : keypoints.shape[0]] = keypoints
+            batched_scores[i, : scores.shape[0]] = scores
+            batched_descriptors[i, : descriptors.shape[0]] = descriptors
+            batched_mask[i, : scores.shape[0]] = 1
+
+        keypoints = batched_keypoints
+        scores = batched_scores
+        descriptors = batched_descriptors
+        mask = batched_mask
+        last_hidden_state = batched_last_hidden_state
+        if output_hidden_states:
+            hidden_states = batched_hidden_states
+        else:
+            hidden_states = None
+        if not return_dict:
+            return tuple(
+                v for v in [keypoints, scores, descriptors, mask, last_hidden_state, hidden_states] if v is not None
+            )
+
+        return ImagePointDescriptionOutput(
+            keypoints=keypoints,
+            scores=scores,
+            descriptors=descriptors,
+            mask=mask,
+            last_hidden_state=last_hidden_state,
+            hidden_states=hidden_states,
+        )
+
+
+@add_start_docstrings(
+    "SuperPoint model outputting keypoints and descriptors.",
+    SUPERPOINT_START_DOCSTRING,
+)
+class SuperPointModelForInterestPointDescription(SuperPointPreTrainedModel):
```

Comment:
it should be named `SuperPointForInterestPointDescription`
Please, check all occurrences and replace `SuperPointModelForInterestPointDescription` by `SuperPointForInterestPointDescription`

Reply:
Fixed in [this commit](https://github.com/huggingface/transformers/pull/25786/commits/a8f7e1c97b89d563552c0e32c350eb7ebead1506)